### ISSUE TITLE
What happens if the cleanup passed to JSWeakFactory is not a function?

### ISSUE THREAD

**Issue by marjakh** [tc39/proposal-weakrefs#37]
... when do we detect it?

**Comment by marjakh** [comment]
Hmm, should've said "is not callable".

**Comment by tschneidereit** [comment]
Agreed, that should fail as early as possible.

**Comment by littledan** [comment]
This is fixed in @gsathya 's spec text:

> If Type(cleanupCallback) is not Callable, throw a TypeError exception.

I think we should reword this as "If IsCallable(cleanupCallback) is false", as Callable is not a type, but the spirit is there.
